execution/src/execute.rs

```diff
@@ -119,36 +122,41 @@ pub(super) fn execute_script<G: GasMeter>(
         args,
         gas_meter,
         traversal_context,
+        code_storage,
     )?;
     Ok(())
 }
 
-pub(super) fn execute_l2_contract<G: GasMeter>(
+// TODO(#329): group MoveVM elements (session, traversal_context,
+// gas_meter, module_storage) together.
+#[allow(clippy::too_many_arguments)]
+pub(super) fn execute_l2_contract<G: GasMeter, MS: ModuleStorage>(
     signer: &AccountAddress,
     contract: &AccountAddress,
     value: U256,
     data: Vec<u8>,
     session: &mut Session,
     traversal_context: &mut TraversalContext,
     gas_meter: &mut G,
+    module_storage: &MS,
 ) -> moved_shared::error::Result<Vec<Log<LogData>>> {
     let module = ModuleId::new(EVM_NATIVE_ADDRESS, EVM_NATIVE_MODULE.into());
     let function_name = EVM_CALL_FN_NAME;
     // Unwraps in serialization are safe because the layouts match the types.
-    let args = vec![
-        Value::address(*signer)
-            .simple_serialize(&ADDRESS_LAYOUT)
-            .unwrap(),
-        Value::address(*contract)
-            .simple_serialize(&ADDRESS_LAYOUT)
-            .unwrap(),
-        Value::u256(value.to_move_u256())
-            .simple_serialize(&U256_LAYOUT)
-            .unwrap(),
-        Value::vector_u8(data)
-            .simple_serialize(&CODE_LAYOUT)
-            .unwrap(),
-    ];
+    let args: Vec<Vec<u8>> = [
+        (Value::address(*signer), &ADDRESS_LAYOUT),
+        (Value::address(*contract), &ADDRESS_LAYOUT),
+        (Value::u256(value.to_move_u256()), &U256_LAYOUT),
+        (Value::vector_u8(data), &CODE_LAYOUT),
+    ]
+    .into_iter()
+    .map(|(value, layout)| {
+        ValueSerDeContext::new()
```

Comment:
Maybe we should implement a `serialize` helper method. Not necessary, but it could be useful.

Reply:
Yeah, it might be nice to have a helper trait for this now that it's a little more verbose. But that can be done in a separate PR.